src/main/java/edu/kit/kastel/sdq/artemis4j/grading/git/VCSTokenCloningStrategy.java

```diff
@@ -0,0 +1,55 @@
+/* Licensed under EPL-2.0 2024. */
+package edu.kit.kastel.sdq.artemis4j.grading.git;
+
+import java.time.ZonedDateTime;
+
+import edu.kit.kastel.sdq.artemis4j.ArtemisNetworkException;
+import edu.kit.kastel.sdq.artemis4j.client.UserDTO;
+import edu.kit.kastel.sdq.artemis4j.grading.ArtemisConnection;
+import org.eclipse.jgit.api.CloneCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Cloning strategy that uses a personal access token (PAT) to clone a repository.
+ */
+public class VCSTokenCloningStrategy implements CloningStrategy {
+    private static final Logger log = LoggerFactory.getLogger(VCSTokenCloningStrategy.class);
+
+    private final CredentialsProvider credentialsProvider;
+
+    public VCSTokenCloningStrategy(String tokenOverride, ArtemisConnection connection) throws ArtemisNetworkException {
+        var assessor = connection.getAssessor();
+
+        String token;
+        if (tokenOverride != null) {
+            token = tokenOverride;
+        } else if (assessor.getGitToken().isPresent()) {
+            token = assessor.getGitToken().get();
```

Comment:
Also check that the token does not getting invalid in the next X (e.g., 2) days. If so, regenerate the token.